packages/provider/provider/index.js

```diff
@@ -82,11 +82,11 @@ export default class Provider extends Component {
   saveState(state, fn) {
     if (!this.mounted) return debug('root component no longer mounted');
 
-    this.setState(state, () => {
+    this.setState(state, function () {
       if (!this.mounted) return debug('root component no longer mounted');
 
       fn(...arguments);
```

Comment:
If you're addressing `...args` below, shouldn't this also be changed to `fn.apply(this, arguments)`?

Reply:
You'd think that... except babel is totally fine with this. it turns it into that anyway. I could do it for it. but yeah, there's weird caveats with when/how array spread breaks :(

Reply:
basically....
![flip](https://media.giphy.com/media/MXWpHJhkIDFU4/giphy.gif)

Reply:
![yaybabel](https://user-images.githubusercontent.com/28071/53276306-d7b3f900-36fe-11e9-9d37-2d684be724ea.gif)